### ISSUE TITLE
"status" field of transaction receipts is returning wrong value sometimes

### ISSUE THREAD

**Issue by wmitsuda** [ledgerwatch/erigon#2236]
I noticed in the eth_getTransactionReceipt the status field is returning wrong value for some transactions, I noticed it is returning "0" indicating a revert for many transactions that were actually successful ("1")

You can reproduce it by getting the receipt from many public node providers, for this example I choose a random tx "0x60ed161964624fb79abff5e3ae968f0b48c47a2f2898dbc8207dbd625f4b6e15":

`curl https://mainnet.infura.io/v3/<key> \
    -X POST \
    -H "Content-Type: application/json" \
    -d '{"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params": ["0x60ed161964624fb79abff5e3ae968f0b48c47a2f2898dbc8207dbd625f4b6e15"],"id":1}' | jq`

Infura returns: "status": "0x1"

`curl http://api.archivenode.io/<key>/erigon \
    -X POST \
    -H "Content-Type: application/json" \
    -d '{"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params": ["0x60ed161964624fb79abff5e3ae968f0b48c47a2f2898dbc8207dbd625f4b6e15"],"id":1}' | jq`

Archivenode.io pinned to erigon returns: "status": 0, like my personal erigon node.

`curl https://eth-mainnet.alchemyapi.io/v2/<key> \
    -X POST \
    -H "Content-Type: application/json" \
    -d '{"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params": ["0x60ed161964624fb79abff5e3ae968f0b48c47a2f2898dbc8207dbd625f4b6e15"],"id":1}' | jq`

Alchemy returns: "status": "0x1"

Etherscan returns success, which corresponds to "status": "0x1"

I'm using the v2021.06.05 tag.

**Comment by alysak6075** [comment]
I am having the same issue.

**Comment by metala** [comment]
https://github.com/ledgerwatch/erigon/commit/629205650

**Comment by AlexeyAkhunov** [comment]
The correct receipts have been broken by the migration, unfortunately.

Please use this workaround to fix your receipts (will be merged into devel branch soon), PR is here: #2284

```
make hack
./build/bin/hack --action scanReceipts --chaindata <datadir>/erigon/chaindata --block 12600000
```
it checks 100k blocks after given one, and every 30s it commits the changes and prints progress like this:
```
INFO [07-03|13:42:59.728] Commit                                   block=12602846 fixed=583
INFO [07-03|13:43:29.802] Commit                                   block=12603773 fixed=1146
INFO [07-03|13:43:59.724] Commit                                   block=12604261 fixed=1508
INFO [07-03|13:44:29.760] Commit                                   block=12604789 fixed=1856
INFO [07-03|13:44:59.749] Commit                                   block=12605326 fixed=2160
INFO [07-03|13:45:29.789] Commit                                   block=12605859 fixed=2499
INFO [07-03|13:45:59.783] Commit                                   block=12606402 fixed=2826
```

**Comment by AskAlexSharov** [comment]
closing because looks done. feel free to re-open
